driver/driver.go

```diff
@@ -92,26 +96,52 @@ func (d driver) Open(dsn string) (sqldriver.Conn, error) {
 	}
 
 	if ci.standardDSN {
-		if ci.params["ssl"] != nil {
-			tlsConfigName := ci.params.Get("ssl")
-			switch tlsConfigName {
-			case "true":
-				// This actually does insecureSkipVerify
-				// But not even sure if it makes sense to handle false? According to
-				// client_test.go it doesn't - it'd result in an error
-				c, err = client.Connect(ci.addr, ci.user, ci.password, ci.db, func(c *client.Conn) { c.UseSSL(true) })
-			case "custom":
-				// I was too concerned about mimicking what go-sql-driver/mysql does which will
-				// allow any name for a custom tls profile and maps the query parameter value to
-				// that TLSConfig variable... there is no need to be that clever.
-				// Instead of doing that, let's store required custom TLSConfigs in a map that
-				// uses the DSN address as the key
-				c, err = client.Connect(ci.addr, ci.user, ci.password, ci.db, func(c *client.Conn) { c.SetTLSConfig(customTLSConfigMap[ci.addr]) })
-			default:
-				return nil, errors.Errorf("Supported options are ssl=true or ssl=custom")
+		var timeout time.Duration
+		configuredOptions := make([]client.Option, 0, len(ci.params))
+		for key, value := range ci.params {
+			if key == "ssl" && len(value) > 0 {
+				tlsConfigName := value[0]
+				switch tlsConfigName {
+				case "true":
+					// This actually does insecureSkipVerify
+					// But not even sure if it makes sense to handle false? According to
+					// client_test.go it doesn't - it'd result in an error
+					configuredOptions = append(configuredOptions, UseSslOption)
+				case "custom":
+					// I was too concerned about mimicking what go-sql-driver/mysql does which will
+					// allow any name for a custom tls profile and maps the query parameter value to
+					// that TLSConfig variable... there is no need to be that clever.
+					// Instead of doing that, let's store required custom TLSConfigs in a map that
+					// uses the DSN address as the key
+					configuredOptions = append(configuredOptions, func(c *client.Conn) error {
+						c.SetTLSConfig(customTLSConfigMap[ci.addr])
+						return nil
+					})
+				default:
+					return nil, errors.Errorf("Supported options are ssl=true or ssl=custom")
+				}
+			} else if key == "timeout" && len(value) > 0 {
+				if timeout, err = time.ParseDuration(value[0]); err != nil {
+					return nil, errors.Wrap(err, "invalid duration value for timeout option")
+				}
+			} else {
+				if option, ok := options[key]; ok {
```

Comment:
Do you think we can expose a `SetDSNOptions` to let user add their own modifications to `Conn`?

Reply:
I've added `SetDsnOptions()`, hopefully I've understood your request and implemented as you intended.